### PR TITLE
Add a CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by cgwalters**
Motivated by improving our project standards.

### CODE REVIEW

**Review by giuseppe — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+## Development environment
+
+Composefs at its core is a low-dependency C library and CLI tools. You'll
+need a Linux environment, which could be a container or a VM/physical system.
+
+composefs should be buildable on nearly any relatively modern Linux OS/distribution.
+
+### Building and testing
+
+composefs uses [meson](https://mesonbuild.com/) - please
```
Comment (giuseppe) [inline CONTRIBUTING.md:25]: noticed afterwards.  Does it miss anything at the end?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+If you set your `user.name` and `user.email` git configs, you can sign your
+commit automatically with `git commit -s`.
+
+### Git commit style
+
+Please look at `git log` and match the commit log style, which is very
+similar to the
+[Linux kernel](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git).
+
+You may use `Signed-off-by`, but we're not requiring it.
```
Comment (giuseppe) [inline CONTRIBUTING.md:114]: I think the DCO bot checks for the `Signed-off-by` line

**Comment by giuseppe** [comment]
used this PR as a base for https://github.com/containers/crun/pull/1590

**Comment by eriksjolund** [comment]
Sidenote regarding the line

https://github.com/containers/composefs/blob/0cb1f05a9fdc6a00a2628ed931d0c80f801e22ac/CONTRIBUTING.md?plain=1#L128

No previous commits fulfil the requirement

```
$ git clone -q https://github.com/containers/composefs.git
$ git -C composefs log | grep 'Closes: #' | wc -l
       0
```

(I'm fine with the change. I just wanted to mention this fact)

**Pull request merged**